### PR TITLE
Add PAPI functionality to Dr Hook

### PR DESCRIPTION
**Pull request by ioanhadade**

### CODE REVIEW

File: src/fiat/drhook/drhook_papi.c [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+  int prof_papi_numcntrs;
+  bool failed=false;  
+#pragma omp parallel default(none) shared(drhook_papi_rank,drhook_max_counter_name,hookCounters,drhook_papi_event_set,nthreads,prof_papi_numcntrs,failed)
+  {
+    int paperr2,counter;
+    int * events;
+    int number;
+    int j=1;
+#ifdef _OPENMP
+    j=safe_thread_num();
```
Comment (wdeconinck) [inline src/fiat/drhook/drhook_papi.c:220]: `safe_thread_num()` normally returns 0 when `_OPENMP` is not defined.
So I thought it seems redundant to have `j=safe_thread_num()` wrapped with `#ifdef _OPENMP`.
However then the default value `j=1` would become `j=0`.
That would bypass the further `j==k` condition when `nthreads=1` as is the case without `_OPENMP` defined.
Is that the intention, why `j=1`?

**Comment by wdeconinck** [comment]
For sake of development, add `C` to the COMPONENTS in this line:
https://github.com/ecmwf-ifs/fiat/pull/18/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR26

**Comment by MartynF** [comment]
Did fiat-print-bindings require OpenMP in C? It looks like it has pragmas!

**Comment by MartynF** [comment]
> @MartynF any occurrence of `long_long` should be replaced with `long long`. See https://github.com/icl-utk-edu/papi/blob/master/src/papi.h#L554-L559

In version 7.... which isn't distributed with majority use OS's including current HPC right now ....

**Comment by ioanhadade** [comment]
I would like to get this merged as it has been a while.
What is the holdup @wdeconinck?

**Comment by wdeconinck** [comment]
> I would like to get this merged as it has been a while. What is the holdup @wdeconinck?

I will take this PR up further now.

**Comment by wdeconinck** [comment]
@ioanhadade @MartynF I have created a new PR #26  from the latest develop branch. Please check it still satisfies, and let's continue discussion there. I will close this PR at the moment. Thanks @MartynF for this nice feature!

**Pull request closed without merging**